packages/superset-ui-legacy-preset-chart-nvd3/src/utils.js

```diff
@@ -58,29 +60,34 @@ export function drawBarValues(svg, data, stacked, axisFormat) {
         })
       : [];
 
-  const groupLabels = svg.select('g.nv-barsWrap').append('g');
-
-  svg
-    .selectAll('g.nv-group')
-    .filter((d, i) => !stacked || i === countSeriesDisplayed - 1)
-    .selectAll('rect')
-    .each(function each(d, index) {
-      const rectObj = d3.select(this);
-      if (rectObj.attr('class').includes('positive')) {
-        const transformAttr = rectObj.attr('transform');
-        const yPos = parseFloat(rectObj.attr('y'));
-        const xPos = parseFloat(rectObj.attr('x'));
-        const rectWidth = parseFloat(rectObj.attr('width'));
-        const textEls = groupLabels
-          .append('text')
-          .attr('y', yPos - 5)
-          .text(format(stacked ? totalStackedValues[index] : d.y))
-          .attr('transform', transformAttr)
-          .attr('class', 'bar-chart-label');
-        const labelWidth = textEls.node().getBBox().width;
-        textEls.attr('x', xPos + rectWidth / 2 - labelWidth / 2); // fine tune
-      }
-    });
+  svg.selectAll('.bar-chart-label').remove();
+  setTimeout(() => {
+    const groupLabels = svg
+      .select('g.nv-barsWrap')
+      .append('g')
```

Comment:
This line still adds new `g`

Reply:
Sorry, I don't know enough about SVG to know if this is bad. Is the problem that we're adding a `g`, or that every call will add **another** `g`? Because since it has the `'bar-chart-label` class, it will get removed next time (see line 63), so we'll add at most one `g`.

Reply:
I took a closer look. This is ok, but let's use another `class` name that is not `bar-chart-label` e.g. `bar-chart-label-group` because this will be the same with the `text` in `.each()` below

Reply:
Cool, let me fix this. Thanks, @kristw!